### DISCUSSION TITLE
Excessive CPU usage

### DISCUSSION THREAD

**Discussion by jdanna** [greghesp/ha-bambulab#369] [General]
Hi - I’ve got this setup with my new A1, and it works great - except after awhile the CPU usage (checked with py-spy on the HA process) - for the the bambu integration spikes to 100% and stays there.

Is there a log somewhere I can look at to try and narrow down what’s going on? For now I just removed the integration to make my HA instance usable again.

**Comment by AdrianGarside** [comment]
Does the debug log show it doing something very repetitive - more than just activity every ~ second but instead something like activity every few 10s of milliseconds? Also, what version of the integration are you running?

**Reply by jdanna** [reply]
Yea thats where im confused - i enabled debug logging but im not sure where to look to actually see that log?

**Comment by AdrianGarside** [comment]
If you enabled debug logging for just the integration then you can turn it off again to trigger a download of the logs to the browser. But we recently discovered that turning on the debug logs that way only enables debug logging for half of the integration code. If you turn on debug logging globally via the HA configuration.yaml or just for the integration we get the full logs and I suspect we'll need them in this this case:

What you add to configuration.yaml:

logger:
  default: info
  logs:
    custom_components.bambu_lab: debug

But to get at the resultant logs needs access to the home assistant working directory and how you can do that will depend how you have your home assistant set up. I have mine running in docker with a config directory projected into my home directory so I just go access ~\.homeassistant\home-assistant.log

But I think you can get the most recent log file via:
Settings->System->Logs->Download full logs

**Reply by jdanna** [reply]
Ok cool - I guess the good news (or bad news) is that I uninstalled and reinstalled the integration (it’s now one dot version newer then what I had) - and so far it hasn’t cropped back up.

no other changes on the printer or HA config.

I’ll keep an eye on it and if it starts aagain I’ll get the logs 

thanks

**Reply by AdrianGarside** [reply]
When you hit this, was the chamber image updating correctly or had it got stuck on a stale image?

**Comment by AdrianGarside** [comment]
Tracking this in #384. I've been hitting it a bunch and seeing the combo symptoms that the chamber image stops updating and python spikes to 100% cpu usage. I've rewritten the chamber image processing to be a lot simpler and more efficient but since I couldn't come up wtih a theory for how it gets itself into a tight loop I don't know if it'll actually fix this bug. New dev release with that change:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.5-dev1

**Reply by jdanna** [reply]
Yea I was just going to reply again that after a few days it started again for me so I disabled the integration. I don’t think the image was updating - but my HA instance was so slow from the CPU usage, i’m not positive.

 I did have debug logging on - but couldn’t find anything relevant in the logs. I’ll try the new version as well.

**Comment by AdrianGarside** [comment]
Fixed in https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.6-dev2
